static/app/views/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -252,10 +253,16 @@ export class SentryAppExternalForm extends Component<Props, State> {
 
     if (fieldToPass.type === 'select') {
       // find the options from state to pass down
-      const defaultOptions = (field.choices || []).map(([value, label]) => ({
+      let defaultOptions = (field.choices || []).map(([value, label]) => ({
         value,
         label,
       }));
+      // there are no choices if skip_load_on_open is set, just use what's in the defaultResetValue
+      if (field.skip_load_on_open === true) {
```

Comment:
There is probably a more elegant way to write this and/or combine with the definition of `defaultOptions` here, but I couldn't get anything else working.

Reply:
You could keep it `const` and have it be a super long ternary 

```js
const defaultOptions = field.skip_load_on_open 
      ? [ {value: defaultResetValues[field.name], label: defaultResetValues[field.name]}] 
      :  (field.choices || []).map(([value, label]) => ({value, label}));
```
But honestly this way is more clear and easier to read. You can change the boolean check to remove `=== true` since that's implied.

Reply:
```suggestion
      if (field.skip_load_on_open) {
```

Reply:
👍  I agree with @leeandher that separating it out makes it more readable